config/internal/ml-metadata/metadata-grpc-tls-config-secret.yaml.tmpl

```
@@ -0,0 +1,23 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ds-pipeline-metadata-grpc-tls-config-secret-{{.Name}}
+  namespace: {{.Namespace}}
+  labels:
+    component: metadata-grpc-server
+stringData:
+  config.proto: |
+    connection_config {
+      mysql {
+        host: "{{.DBConnection.Host}}"
+        port: {{.DBConnection.Port}}
+        database: "{{.DBConnection.DBName}}"
+        user: "{{.DBConnection.Username}}"
+        password: "{{.DBConnection.DecodedPassword}}"
+      }
+    }
+    ssl_config {
+      server_cert: "{{.MlmdGrpcCertificateContents}}"
+      server_key: "{{.MlmdGrpcPrivateKeyContents}}"
+      client_verify: false // true for mTLS
```

small typo: `mTLS`

---

it actually IS for mTLS, which I don't think we're supporting in this PR ... but I'm not all the way through it yet :)

Ref: https://github.com/google/ml-metadata/blob/master/ml_metadata/proto/metadata_store.proto#L869-L870

"client certificate" i.e. mTLS vs regular TLS

---

well it's always hardcoded false, so no, we're not supporting mTLS here.

I'd have probably put
```suggestion
      client_verify: false // controls mTLS, which we don't use, so hardcode to false
```

---

Comment changed as suggested.